regulation-worker/internal/delete/batch/batch.go

```diff
@@ -31,64 +29,35 @@ import (
 
 var (
 	pkgLogger             = logger.NewLogger().Child("batch")
-	regexRequiredSuffix   = regexp.MustCompile(".json.gz$")
 	StatusTrackerFileName = "rudderDeleteTracker.txt"
-	supportedDestinations = []string{"S3"}
+	supportedDestinations = []string{"S3", "S3_DATALAKE"}
 )
 
 const listMaxItem int64 = 1000
 
-type deleteManager interface {
-	delete(ctx context.Context, patternFilePtr, targetFilePtr string) ([]byte, error)
+type PatternDeleter interface {
+	DeletePattern(ctx context.Context, pattern []string, source []byte) ([]byte, error)
 }
 
 type Batch struct {
 	mu         sync.Mutex
 	FM         filemanager.FileManager
-	DM         deleteManager
 	TmpDirPath string
 }
 
-// return appropriate deleteManger based on destination Name
-func getDeleteManager(destName string) (*S3DeleteManager, error) {
-	switch destName {
-	case "S3":
-		return &S3DeleteManager{}, nil
-	default:
-		return nil, model.ErrDestNotImplemented
-	}
-}
-
 // returns list of all .json.gz files.
 // NOTE: assuming that all of batch destination have same file system as S3, i.e. flat.
 func (b *Batch) listFiles(ctx context.Context) ([]*filemanager.FileObject, error) {
 	pkgLogger.Debugf("getting a list of files from destination")
+
+	// TODO: What happens if we have files greater than 1000 ?
 	fileObjects, err := b.FM.ListFilesWithPrefix(ctx, "", "", listMaxItem)
```

Comment:
Valid point, we need to make sure we keep iterating

Reply:
I checked and we do capture the token for next page every-time we list the maximum files first time.